bot/converters.py

```diff
@@ -267,37 +268,60 @@ async def convert(self, ctx: Context, arg: str) -> int:
         return snowflake
 
 
-class SourceConverter(Converter):
-    """Convert an argument into a help command, tag, command, or cog."""
+class SourceTransformer(app_commands.Transformer):
+    """Transform an argument into a tag, command or cog."""
 
-    @staticmethod
-    async def convert(ctx: Context, argument: str) -> SourceType:
-        """Convert argument into source object."""
-        if argument.lower() == "help":
-            return ctx.bot.help_command
+    async def transform(self, interaction: discord.Interaction, cog_command_or_tag: str) -> SourceType:
+        """Transform cog_command_or_tag into source object."""
+        if cog_command_or_tag.lower() == "help":
+            return interaction.client.help_command
 
-        cog = ctx.bot.get_cog(argument)
+        cog = interaction.client.get_cog(cog_command_or_tag)
         if cog:
             return cog
 
-        cmd = ctx.bot.get_command(argument)
+        cmd = interaction.client.get_command(cog_command_or_tag)
         if cmd:
             return cmd
 
-        tags_cog = ctx.bot.get_cog("Tags")
+        slash_cmd = interaction.client.tree.get_command(cog_command_or_tag)
+        slash_cmd = slash_cmd if not isinstance(slash_cmd, app_commands.ContextMenu) else None
+        if slash_cmd:
+            return slash_cmd
+
+        tags_cog = interaction.client.get_cog("Tags")
         show_tag = True
 
         if not tags_cog:
             show_tag = False
         else:
-            identifier = TagIdentifier.from_string(argument.lower())
+            identifier = TagIdentifier.from_string(cog_command_or_tag.lower())
             if identifier in tags_cog.tags:
                 return identifier
-        escaped_arg = escape_markdown(argument)
+        escaped_arg = escape_markdown(cog_command_or_tag)
 
         raise BadArgument(
```

Comment:
It would be handled by the error handler because any error raised in the transformer raises `TransformerError`. The reason I did this was because I wanted to pass the error message to the error handler through the `cause` dunder. Rasining `TransformerError` will only add complexity as you can't pass your own custom error message to it.

Reply:
Oh, I didn't know that the command would be enclosed with `TransformerError.

Yeah `TransformerError` doesn't allow you to do that much, even though you could bend around it by passing the error messafe to the `opt_type` parameter.

I just don't like the fact that we'd be throwing a test command error from a slash one.

Reply:
I don't know if theres a better way maybe we can raise `ValueError` instead?

Reply:
Updated in https://github.com/python-discord/bot/pull/2446/commits/6059dd077468b1338109327793400469ed96096e